### ISSUE TITLE
Add XGBoost Multi-class classification inference support (output transformation & output tree aggregation)

### ISSUE THREAD

**Issue by yanndupis** [tf-encrypted/moose#640]
## Context 

related to #633

For multi-class classification task, XGBoost model expects to return the predicted class and also the probability associated with each class. See the following example:

```
from xgboost import XGBClassifier
from sklearn.datasets import load_iris

data = load_iris()

X = data.data[((data.target==0) | (data.target==1) | (data.target==2))]
y = data.target[((data.target==0) | (data.target==1) | (data.target==2))]

model = XGBClassifier(n_estimators=3, max_depth=2)
model.fit(X, y)

print(model.predict([[6.3, 3.4, 3, 0.5]])) # return [1]
model.predict_proba([[6.3, 3.4, 3, 0.5]]) # return [0.15864462, 0.68252754, 0.15882783]
```

For muli-class classification, XGBoost uses the [ones-vs-rest](https://scikit-learn.org/stable/modules/generated/sklearn.multiclass.OneVsRestClassifier.html) approach where we train a model for each class. So if we train an XGBoost model limited to 1 tree (n_estimators=1), if we try to predict 3 different classes, we will end up 3 different decision trees (one for each class). So when aggregating the weights (as for Regression and binary classification) from each decision tree, we need to group/aggregate the decision trees by class.

## Objectives
- Validate how the aggregation of the outputs of the ensemble of decision trees should be aggregated then transformed to return predicted class and probability associated to each each class. 
- Validate if we need `softmax` to return probabilities. 
- Extend the EDSL computation developed in the context of regression (#633, #637) task to return predicted class and probability for each class.
